### PR TITLE
Add dal.hpp

### PR DESCRIPTION
**Pull request by rlnx**
- Add global header that includes all library functionality
- Add test that verifies all algorithms are include in `dal.hpp`. There is no such checks for tables and IO, expect they are not updated so frequently
- Move `.clang-format` one level up, update paths in CI scripts

### CODE REVIEW

**Review by napetrov — APPROVED** [review]
Looks good

File: cpp/oneapi/dal.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/* Algos */
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/algo/kmeans.hpp"
+#include "oneapi/dal/algo/kmeans_init.hpp"
+#include "oneapi/dal/algo/knn.hpp"
+#include "oneapi/dal/algo/linear_kernel.hpp"
+#include "oneapi/dal/algo/pca.hpp"
+#include "oneapi/dal/algo/rbf_kernel.hpp"
+#include "oneapi/dal/algo/svm.hpp"
```
Comment (michael-smirnov) [inline cpp/oneapi/dal.hpp:45]: Do you plan to include graphs?

Comment (rlnx) [reply cpp/oneapi/dal.hpp:45]: Not sure if we have to include preview feature here

Comment (rlnx) [reply cpp/oneapi/dal.hpp:51]: Added

**Pull request merged**